libs/scripts/taxonomy.js

```diff
@@ -84,23 +84,19 @@ function parseTaxonomyJson(data, root) {
         : LEVEL_INDEX.level1);
 
     const name = level3 || level2 || level1;
-
     const category = row[TAXONOMY_FIELDS.type]?.trim().toLowerCase() || INTERNALS;
 
     // skip duplicates
     if (!isProduct(category) && taxonomy.topics[name]) return taxonomy;
     if (isProduct(category) && taxonomy.products[name]) return taxonomy;
 
-    const link = [row[TAXONOMY_FIELDS.link]].reduce((_url) => {
-      if (_url) {
-        const u = new URL(_url);
-        const current = new URL(window.location.href);
-        return `${current.origin}${u.pathname}`;
-      }
-
-      return `${root}/${generateUri(name)}`;
-    }, null);
-
+    const taxLink = row[TAXONOMY_FIELDS.link]
+      ? new URL([row[TAXONOMY_FIELDS.link]])
+      : generateUri(name);
+    const path = taxLink.pathname
+      ? taxLink.pathname?.replace('.html', '').split('/topics/').pop()
+      : taxLink;
+    const link = `${root}/${path}`;
```

Comment:
Conditionally returning a string or URL object might lead to problems in the future, something like this could be better:

```suggestion
    const taxUrl = row[TAXONOMY_FIELDS.link] && new URL([row[TAXONOMY_FIELDS.link]])
    const path = taxUrl.pathname?.replace('.html', '').split('/topics/').pop() || generateUri(name)
    const link = `${root}/${path}`;
```